src/plugins_types/union.c

```diff
@@ -176,9 +176,13 @@ lyplg_type_store_union(const struct ly_ctx *ctx, const struct lysc_type *type, c
         subvalue->orig_len = value_len;
         options &= ~LYPLG_TYPE_STORE_DYNAMIC;
     } else {
-        subvalue->original = calloc(1, value_len);
-        LY_CHECK_ERR_GOTO(!subvalue->original, ret = LY_EMEM, cleanup);
-        memcpy(subvalue->original, value, value_len);
+        if (value_len) {
+            subvalue->original = calloc(1, value_len);
+            LY_CHECK_ERR_GOTO(!subvalue->original, ret = LY_EMEM, cleanup);
+            memcpy(subvalue->original, value, value_len);
+        } else {
+            subvalue->original = "";
+        }
```

Comment:
For the code to be a bit better organized, you could use `if () ... else if () ... else ...`. With the `else if` condition being `value_len` (execute the current code) and `else` with simply `subvalue->original = strdup("");` with error check. That way the member value is unified and always dynamic. You also get rid of one level of depth.